src/processors/sync.rs

```diff
@@ -54,7 +59,25 @@ pub fn process_sync_record<'a>(
     if let Some(load_average) = ix.load_average().take() {
         record.set_load_average(load_average);
     }
-
+    let new_size = borsh::object_length(&record)?;
+    let old_size = data.len();
+    if old_size == new_size {
+        return Ok(record.serialize(&mut *data)?);
+    };
+    drop(data);
```

Comment:
Could be removed

Reply:
Refcell complains about outstanding reference